src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -282,6 +282,34 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
     executeOrResume(backendCall, runIdForResumption = runId)
   }
 
+  def useCachedCall(avoidedTo: BackendCall, backendCall: BackendCall)(implicit ec: ExecutionContext): Future[ExecutionHandle] = Future {
+    val log = workflowLoggerWithCall(backendCall)
+    authenticateAsUser(backendCall.workflowDescriptor) { gcs =>
+      Try(gcs.copyPrefix(avoidedTo.callGcsPath, backendCall.callGcsPath, Option(log))) match {
+        case Success(attempts) if attempts.forall(_.isSuccess) =>
+          postProcess(backendCall) match {
+            case Success(outputs) => SuccessfulExecutionHandle(outputs, backendCall.downloadRcFile.get.stripLineEnd.toInt, backendCall.hash)
+            case Failure(ex: AggregatedException[_]) if ex.exceptions.map(_.exception).isInstanceOf[SocketTimeoutException] =>
+              // TODO: What can we return here to retry this operation?
+              val error = "Socket timeout occurred in evaluating one or more of the output expressions"
+              log.error(error, ex)
+              FailedExecutionHandle(new Throwable(error, ex))
+            case Failure(ex) => FailedExecutionHandle(ex)
+          }
+        case Success(attempts) =>
+          TryUtil.sequence(attempts.toSeq) match {
```

Comment:
Well I would definitely like to be convinced.  I've changed my view before (i.e. `Try` vs. `try`) and I can certainly be convinced again.

I've read a lot about scalaz validations recently and I feel like I get it pretty well, but of course I'm no expert and I might not know what I don't know.  One of the [blog posts](http://www.47deg.com/blog/fp-for-the-average-joe-part-1-scalaz-validation) I found explained a rational behind using `ValidationNEL` instead of `Try`, and I read it top-to-bottom in great detail to make sure I understood every detail.  Even in his example, I think a MUCH better solution would be to ignore scalaz altogether, and just use `TryUtil.sequence` like this

```scala
def toInts(maybeInts : List[String]) : Try[List[Int]] = {
    TryUtil.sequence(maybeInts.map(i => Try(i.toInt)))
}
```